st2common/st2common/util/virtualenvs.py

```diff
@@ -73,6 +74,19 @@ def setup_pack_virtualenv(pack_name, update=False, logger=None, include_pip=True
         msg = 'Pack "%s" is not installed. Looked in: %s' % (pack_name, search_paths)
         raise Exception(msg)
 
+    try:
+        pack_meta = get_pack_metadata(pack_path)
+        has_pack_meta = True
+    except ValueError:
+        # Pack is missing meta file
+        has_pack_meta = False
+
+    if has_pack_meta:
+        logger.debug('Checking pack specific Python version.')
+        if 'python3' in pack_meta.keys():
```

Comment:
A while ago we added a couple more fields to the pack model and pack.yaml file, notably `dependencies` and `system`.

Perhaps it would be better to utilize `system` field instead of a top-level one (e.g.` system.python3 = True` or similar).

We also need to decide and clarify what that attribute really means - does it means that pack supports just Python 3 or that it supports Python 3 in addition to Python 3? And once we decide what it means and represents, we should document it :)

Reply:
Agreed on the documentation. Top-level attribute seems off also, where is the meta file model? I couldn't find it anywhere.
Currently, it means that the pack should use a Python3 virtualenv. So it supports Python 3 at least, but shouldn't fall back to 2

Reply:
We don't have a schema for `system` attribute yet, but you can add it and extend it here - https://github.com/StackStorm/st2/blob/master/st2common/st2common/models/api/pack.py#L135

For now, you can just add declaration for `system.python3` attribute there under `system` field.